src/cryptography/utils.py

```diff
@@ -143,7 +146,7 @@ def inner(instance):
         setattr(instance, cached_name, result)
         return result
 
-    return property(inner)
+    return typing.cast(_F, property(inner))
```

Comment:
Err, this isn't a callable though, it's a property?

Reply:
This is true. Is there a way to propagate the type of `inner` through this decorator? It seems like there must be. I treated it like a generic decorator for this purpose and the right types come through via `reveal_type`, but it's an excellent point that this cast is just a lie.